### PR TITLE
[featrue](expr) check expr return type and block column type 

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: be/src/vec/exprs/vexpr_context.cpp [hunk 1]
```diff
@@ -55,6 +61,39 @@ Status VExprContext::execute(vectorized::Block* block, int* result_column_id) {
...
+    auto st = column_data_type->check_column_type(column_ptr.get());
+    if (!st.ok()) {
+        return Status::InternalError(
+                "VExprContext expr match failed ,"
+                "col name = {} , expr expected type = {} , "
+                "column expected type = {} , "
+                "column is nullable = {} , "
+                "error type msg = {}",
+                data.name, expr_data_type->get_name(), column_data_type->get_name(),
+                column_ptr->is_nullable(), st.msg());
```
Comment (zhangstar333) [inline be/src/vec/exprs/vexpr_context.cpp:95]: column_ptr->dump_structure()

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Pull request closed without merging**